release: v11 (#151)
mia-platform/documentation: docs/release-notes/mia-platform-v11.mdx

### Machine-to-Machine Authentication

Service accounts are an essential component of many IT environments, allowing applications and services to access various resources and perform operations without the need for human intervention.
Mia-Platform Console allows to create and manage service accounts, useful for automating processes that previously had to be done by a human user.
Assign roles and permissions to service accounts lets you to improve speed, reliability and scalability of your CI/CD workflows.
For instance, service accounts are useful for applications that require continuous monitoring, reporting, and data processing, or you can give the permission to trigger automatic deploys, which can significantly streamline the process of deploying code changes to an application.
Service accounts are also potentially great for security. Using service accounts promotes the separation of roles and responsibilities, which reduces the risk of credential misuse or theft.

### Webhooks Management

> Each user has different needs, especially dependent on their business. We believe it is critical to give each user the ability to manage any additional custom flows as certain events occur in the console.
> 
> For this reason, we have introduced the ability for the user to be able to subscribe to specific events in Console via HTTP webhooks via our CMS, so that custom HTTP calls can be configured that will be triggered when the respective event occurs in Console.
> 
> For now, the supported events are:
> 
> -Project Creation
> -Service Creation

Reviewer: ```suggestion
Each user has different needs, especially dependent on their business. We believe it is critical to give each user the ability to manage any additional custom flows as certain events occur in the Console.

For this reason, we have introduced the possibility for the user to subscribe to specific events in Console via HTTP webhooks, through our CMS. With this feature it will be possible to configure custom HTTP calls that will be triggered when the respective event takes place in Console

For now, the supported events are:

*Project Creation  
*Service Creation
```

### New Cluster connection

Mia-Platform v11 supports all the best and most widely used Runtime Services on the market.

> Within the console, the experience of connecting a Cluster has been completely improved so that you can choose your configuration from a predefined set of officially supported Runtime Services divided by vendor for a clearer and more intuitive experience.

Reviewer: ```suggestion
Within the Console, the experience of connecting a Cluster has been completely improved, so that you can choose your configuration from a predefined set of officially supported Runtime Services divided by vendor for a clearer and have a more intuitive experience in their connection.
```

In the Console, we have introduced new sections for managing clusters, providers, machine 2 machine authentication, and webhooks, as well as support for GitHub Actions.
These enhancements offer more comprehensive control and a smoother development experience for users, improving the overall Developer Experience.

In Fast Data, the main innovations include the ability to store data in buckets, no-code configuration, and the introduction of new, ready-to-use connectors.
These features make the product even more versatile and powerful, simplifying data management and integration.

For the Backoffice, we have enriched the product with new features, including the ability to configure the entire backoffice in no-code mode.
These updates streamline daily management, allowing for greater control of pages and sections, and increasing operational efficiency.

Don't miss the opportunity to explore all the incredible new features of Mia-Platform V11 and discover how we can help you transform your digital enterprise, providing an optimized development experience centered on developers' needs.
Mia-Platform V11, Simplify Complexity.

## Console

### Providers Management

> We have developed the Mia-Platform Console so that it can integrates properly with the best and most widely used tools and providers on the market. 
> 
> However, this integration needs a proper governance system, and this feature has just that goal: to make it easier to keep constantly documented and ordered all the Providers you want the console to use. Through the new dedicated section - named "Providers" - inside the "Company Overview" area it will be possible to manage the creation, configuration, and editing of the providers used by Mia-Platform Console (GitProvider, Secret Manager and CI/CD Tools).

Reviewer: ```suggestion
We have developed Mia-Platform Console so that it can integrate properly with the best and most widely used tools and providers on the market. 
Therefore, to better enhance this integration, we have introduced a proper governance system, and this feature has precisely this goal: to make it easier to keep constantly documented and ordered all the Providers you want the Console to use. Through the new dedicated section - named "Providers" - inside the "Company Overview" area it will be possible to manage the creation, configuration, and editing of the providers used by Mia-Platform Console (GitProvider, Secret Manager and CI/CD Tools).
```

---
id: mia-platform-v11-overview
title: Mia-Platform v11
sidebar_label: v11 Overview
image: "img/release-note-link-preview.png"
---

import bannerImageUrl from './img/banner-overview-v11.png'

> We are excited to announce the arrival of Mia-Platform V11, a major release that brings numerous new features and improvements to our key products: Console, Fast Data, and Backoffice.

Reviewer: why not "Mia-Platform Backoffice" then?

Mia-Platform v11 supports all the best and most widely used Runtime Services on the market.

Within the console, the experience of connecting a Cluster has been completely improved so that you can choose your configuration from a predefined set of officially supported Runtime Services divided by vendor for a clearer and more intuitive experience.

### Enhanced Runtime area

The Runtime Area has been enhanced with several new k8s resources: after the introduction of Pods and Namespaces sections, we are now welcoming also the Deployments, Services, Cronjobs, and Jobs sections. You can now monitor all these k8s resources and their relative properties directly on Mia-Platform Console.

### Support GitHub Actions

This new version of the console brings with it a very used and appreciated tool in the market: from now on you can use and configure GitHub Actions as CI/CD Tool in Console so that they are used to deploy your workloads.

### Feature Preview

Keep up with innovation and try out our new Console Features available in Preview exclusively for our Console SaaS users.
Access the Feature Preview section from Console launchbar, enable them and give us precious feedback!

> ## Fast Data

Reviewer 2: ```suggestion
## Mia-Platform Fast Data
```

### Er Schema No Code Configuration

We understand the importance of a user-friendly interface, and that's why we have introduced the Er Schema No Code configuration for Fast Data.

With this new feature, you can easily configure and build your Er Schema directly from the System of Record page. The Er Schema section is now more intuitive and easy to use, making it simpler to define relationships between projections and the rules that connect them to one another.

One of the key advantages of this feature is the drag and drop functionality, which allows you to import projections into a canvas quickly. From there, you can easily define their relationships and create rules that connect them to one another.

Overall, the Er Schema No Code configuration for Fast Data is a game-changer in terms of simplifying the process of configuring services from the console, making it easier than ever to create the Er Schema they need to power their data-driven applications.

Find out more about this functionality in the official documentation! <!---TODO add link to the documentation-->

<!---TODO: add picture-->

### Fast Data Connectors

> We are thrilled to announce one of the most requested feature to Fast Data, featuring the addition of new plug-and-play connectors that make it easier than ever to connect and manage your data.

Reviewer 2: ```suggestion
We are thrilled to announce one of the most requested feature to Mia-Platform Fast Data, featuring the addition of new plug-and-play connectors that make it easier than ever to connect and manage your data.
```

With these new connectors, you can seamlessly integrate your data from a wide variety of sources, including databases, bucket storages and more, leveraging on Debezium CDC and some other hand-made connectors.

We understand that managing data can be a complex and time-consuming process, which is why we integrated these connectors to be as easy to use as possible. With just a few clicks, you can quickly and easily set up connections to your data sources, and start managing your data in a way that makes sense for your business.

> At the heart of Fast Data product is a commitment to helping our customers unlock the full potential of their data. With these new plug-and-play connectors, we are proud to take another step towards achieving this goal, and we look forward to seeing the many ways our customers will use them to drive their businesses forward.

Reviewer 2: ```suggestion
At the heart of Mia-Platform Fast Data product is a commitment to helping our customers unlock the full potential of their data. With these new plug-and-play connectors, we are proud to take another step towards achieving this goal, and we look forward to seeing the many ways our customers will use them to drive their businesses forward.
```

Assigning roles and permissions to Service accounts lets you improve the speed, reliability, and scalability of your CI/CD workflows.
Service accounts may be suitable for those applications that require continuous monitoring, reporting, and data processing; alternatively, you can use them also to streamline the process of deploying code changes to an application, for instance by assigning them permissions to trigger automatic deploys.
Moreover, they can also be great for security concerns: using Service accounts promotes the separation of roles and responsibilities, which reduces the risk of credential misuse or theft.
With Mia-Platform Console you can now create and manage Service accounts, and start to implement automated processes with machine-to-machine communication.

### Webhooks Management

Each user has different needs, especially depending on their business. We believe it is essential to give each user the ability to manage additional custom flows when certain events occur in the Console.

> For this reason, we have introduced the possibility for the user to subscribe to specific events in Console via HTTP webhooks, through our CMS. With this feature it will be possible to configure custom HTTP calls that will be triggered when the respective event takes place in Console

Reviewer 3: ```suggestion
For this reason, we have introduced the possibility for the user to subscribe to specific events in Console via HTTP webhooks, through our CMS. With this feature it will be possible to configure custom HTTP calls that will be triggered when the respective event takes place in Console.
```

For now, the supported events are:

* Project Creation  
* Service Creation

### New Cluster connection

Mia-Platform v11 supports all the best and most widely used Runtime Services on the market.

> Within the Console, the experience of connecting a Cluster has been completely improved, so that you can choose your configuration from a predefined set of officially supported Runtime Services divided by vendor and have a clearer and more intuitive experience.

Reviewer 3: ```suggestion
Within the Console, the experience of connecting a Cluster has been significantly improved, so that you can choose your configuration from a predefined set of officially supported Runtime Services divided by vendor and have a clearer and more intuitive experience.
```

---
id: mia-platform-v11-overview
title: Mia-Platform v11
sidebar_label: v11 Overview
image: "img/release-note-link-preview.png"
---

import bannerImageUrl from './img/banner-overview-v11.png'

> We are excited to announce the arrival of Mia-Platform V11, a major release that brings numerous new features and improvements to our key products: Console, Fast Data, and Backoffice.

Reviewer 3: ```suggestion
We are excited to announce the launch of Mia-Platform V11, a major release that brings several new features and improvements to Mia-Platform key products: Console, Fast Data, and Backoffice.
```

This version is designed to provide more advanced and comprehensive solutions in the field of Internal Developer Platforms, Digital Integration Hubs, and Backoffice, with a specific focus on enhancing the Developer Experience.

> In the Console, we have introduced new sections for managing clusters, providers, machine 2 machine authentication, and webhooks, as well as support for GitHub Actions.

Reviewer 3: ```suggestion
In Mia-Platform Console, we have introduced clusters, providers and webhooks management, as well as support for machine-to-machine authentication and GitHub Actions.
```

---
id: mia-platform-v11-overview
title: Mia-Platform v11
sidebar_label: v11 Overview
image: "img/release-note-link-preview.png"
---

import bannerImageUrl from './img/banner-overview-v11.png'

> We are excited to announce the arrival of Mia-Platform V11, a major release that brings numerous new features and improvements to our key products: Console, Fast Data, and Backoffice.
> This version is designed to provide more advanced and comprehensive solutions in the field of Internal Developer Platforms, Digital Integration Hubs, and Backoffice, with a specific focus on enhancing the Developer Experience.

Reviewer: ```suggestion
We are excited to announce the arrival of Mia-Platform V11, a major release that brings several new features and improvements to our key products: Console, Fast Data, and Backoffice.
This version is designed to provide more advanced and comprehensive solutions in the field of Internal Developer Platforms, Digital Integration Hubs, and Backoffice, with a specific focus on enhancing the Developer Experience.
```

In the Console, we have introduced new sections for managing clusters, providers, machine 2 machine authentication, and webhooks, as well as support for GitHub Actions.
These enhancements offer more comprehensive control and a smoother development experience for users, improving the overall Developer Experience.

> In Fast Data, the main innovations include the ability to store data in buckets, no-code configuration, and the introduction of new, ready-to-use connectors.

Reviewer: ```suggestion
In Fast Data, the main innovations include the ability to store data in buckets, No Code configuration, and the introduction of new, ready-to-use connectors.
```

These features make the product even more versatile and powerful, simplifying data management and integration.

For the Backoffice, we have enriched the product with new features, including the ability to configure the entire backoffice in no-code mode.
These updates streamline daily management, allowing for greater control of pages and sections, and increasing operational efficiency.

> Don't miss the opportunity to explore all the incredible new features of Mia-Platform V11 and discover how we can help you transform your digital enterprise, providing an optimized development experience centered on developers' needs.

Reviewer: ```suggestion
Don't miss the opportunity to explore all the incredible new features of Mia-Platform v11 and discover how we can help you transform your digital enterprise, providing an optimized development experience centered on developers' needs.
```

Mia-Platform V11, Simplify Complexity.

## Console

### Providers Management

### Machine-to-Machine Authentication

> Service accounts are an essential component of many IT environments, allowing applications and services to access various resources and perform operations without the need for human intervention.
> Mia-Platform Console allows to create and manage service accounts, useful for automating processes that previously had to be done by a human user.
> Assign roles and permissions to service accounts lets you to improve speed, reliability and scalability of your CI/CD workflows.
> For instance, service accounts are useful for applications that require continuous monitoring, reporting, and data processing, or you can give the permission to trigger automatic deploys, which can significantly streamline the process of deploying code changes to an application.
> Service accounts are also potentially great for security. Using service accounts promotes the separation of roles and responsibilities, which reduces the risk of credential misuse or theft.

Reviewer: ```suggestion
Service accounts are an essential component of many IT environments, as they allow applications and services to access various resources and perform operations without the need for human intervention.
With Mia-Platform Console you can create and manage service accounts, and start automating processes that used to be performed by a human user.
Assigning roles and permissions to service accounts lets you improve the speed, reliability, and scalability of your CI/CD workflows.
For instance, service accounts are useful for applications that require continuous monitoring, reporting, and data processing, or you can give the permission to trigger automatic deploys, which can significantly streamline the process of deploying code changes to an application.
Service accounts are also potentially great for security. Using service accounts promotes the separation of roles and responsibilities, which reduces the risk of credential misuse or theft.
```

### Webhooks Management

### New Cluster connection

### Enhanced Runtime area

> Runtime Area has been enhanced with lots new k8s resources to be monitored directly from Mia-Platform Console.
> After Pods and Namespaces, welcome to Deployments, Services, Cronjobs and Jobs sections, wherein it is possible to monitor these k8s resources and their relative properties.

Reviewer: ```suggestion
The Runtime Area has been enhanced with several new k8s resources, which can now be monitored directly from Mia-Platform Console.
After Pods and Namespaces sections, we are now welcoming the Deployments, Services, Cronjobs, and Jobs sections. You can now monitor all these k8s resources and their relative properties directly on Mia-Platform Console.
```

---
id: v11-overview
title: Mia-Platform v11
sidebar_label: v11 Overview
image: "img/release-note-link-preview.png"
---

> import bannerImageUrl from './img/TODO-REPLACE-IMAGE-banner-overview-v11.png'

Author: This banner must be updated!

We are excited to announce the arrival of Mia-Platform v11, a major release that brings numerous new features and improvements to our key products: Mia-Platform Console, Fast Data, and Backoffice.
This version is designed to provide more advanced and comprehensive solutions in the field of Internal Developer Platforms, Digital Integration Hubs, and Backoffice, with a specific focus on enhancing the Developer Experience.

> In the Console, we have introduced new sections for managing clusters, providers, machine2machine authentication, and webhooks, as well as support for GitHub Actions.

Reviewer 2: ```suggestion
In Mia-Platform Console, we have introduced new sections for managing clusters, providers, machine2machine authentication, and webhooks, as well as support for GitHub Actions.
```

These enhancements offer more comprehensive control and an even more frictionless development experience for users, improving the overall Developer Experience.

In Mia-Platform Fast Data, the main innovations include the ability to store data in buckets, No Code configuration, and the introduction of new, ready-to-use connectors.
These features make the product even more versatile and powerful, simplifying data management and integration.

For Backoffice, we have enriched the product with new features, including the ability to configure it in No Code mode.
These updates streamline daily management, allowing for greater control of pages and sections, and increasing operational efficiency.

Don't miss the opportunity to explore all the incredible new features of Mia-Platform v11 and discover how we can help you transform your digital enterprise, providing an optimized development experience centered on developers' needs.

> Mia-Platform v11, Simplify Complexity.

Reviewer 2: I think we can remove it. It wants to sound like a statement, but in this context, it doesn't work.

> Mia-Platform v11, Simplify Complexity.

Author: ```suggestion
```

## Console

### Providers Management

We have developed Mia-Platform Console so that it can integrate properly with the best and most widely used tools and providers on the market. 

> Therefore, to better enhance this integration, we have introduced a proper governance system, and this feature has precisely this goal: to make it easier to keep constantly documented and ordered all the Providers you want the Console to use. Through the new dedicated section - named "Providers" - inside the "Company Overview" area it will be possible to manage the creation, configuration, and editing of the providers used by Mia-Platform Console (GitProvider, Secret Manager and CI/CD Tools).

Reviewer 2: ```suggestion
Therefore, to better enhance this integration, we have introduced a proper governance system, and this feature has precisely this goal: to make it easier to keep constantly documented and ordered all the Providers you want the Console to use. Through the new dedicated section - named "Providers" - inside the "Company Overview" area it will be possible to manage the creation, configuration, and editing of the providers used by Mia-Platform Console (Git Provider, Secret Manager and CI/CD Tools).
```

Reviewer 2: should it be "provider**s**?

Author: no it is correct "provider" is the type of provider, the actual type might just be "Git" since the others are "Secret Manager" and not "Secret Manager Provider" however they are always referred to as "GIt Provider" so it's fine, I guess.

![Service Accounts Management](./img/overview_v11/service-accounts.png)

### Webhooks Management

Each user has different needs, especially depending on their business. We believe it is essential to give each user the ability to manage additional custom flows when certain events occur in the Console.

For this reason, we have introduced the possibility for the user to subscribe to specific events in Console via HTTP webhooks, through our CMS. With this feature it will be possible to configure custom HTTP calls that will be triggered when the respective event takes place in Console.

For now, the supported events are:

* Project Creation  
* Service Creation

### New Cluster connection

Mia-Platform v11 supports all the best and most widely used Runtime Services on the market.

Within the Console, the experience of connecting a Cluster has been significantly improved, so that you can choose your configuration from a predefined set of officially supported Runtime Services divided by vendor and have a clearer and more intuitive experience.

### Enhanced Runtime area

> The Runtime Area has been enhanced with several new k8s resources: after the introduction of Pods and Namespaces sections, we are now welcoming also the Deployments, Services, Cronjobs, and Jobs sections. You can now monitor all these k8s resources and their relative properties directly on Mia-Platform Console.

Reviewer: ```suggestion
The Runtime Area has been enriched by adding several new k8s resources: after the introduction of Pods and Namespaces sections, we are now welcoming also the Deployments, Services, Cronjobs, and Jobs sections! You can now monitor all these k8s resources and their related properties directly on Mia-Platform Console.
```

![Enhanced Runtime Area](./img/overview_v11/runtime-resources.png)

### Support GitHub Actions

Console v11 supports a widely used and appreciated tool in the market: from now on you can use and configure GitHub Actions as CI/CD Tool in Console to deploy your workloads.

### Feature Preview

> Keep up with innovation and try out our new Console Features available in Preview exclusively for our Console SaaS users.
> Access the Feature Preview section from the Console launchbar, activate them, and give us valuable feedback!
> 

Reviewer: ```suggestion
We want you to always keep up with our innovation process. For this reason, for our Console SaaS users we have introduced the Feature Preview section: a place inside of Mia-Platform Console where you can decide to try out the Console features available in Preview.  

You can access this section from the Console launchbar: to discover more, just visit the Feature Preview section, try our new features and tell us what you think by providing us precious feedback!
```

Re-ingesting data has never been easier with the Bucket Storage Support feature. Users can now easily re-ingest their data into the system, which means they can restore data previously removed with few actions.
Lastly, the Bucket Storage Support feature makes it possible to offload the projections storage. This means that users can now offload their projections relying on the data stored in this separate storage bucket, which reduces the amount of storage needed in the main system. This can lead to significant cost savings and improved system performance.
The Bucket Storage Support feature provides users with greater control over their data and makes it easier to backup, re-ingest, and manage their data.

Do not miss out the [official documentation](/fast_data/bucket_storage_support/overview.md)!

> ## Marketplace

Author: we have an empty title here, is it supposed to hold something?
